scripts/zkai_native_seq32_attention_mlp_predecommit_opening_policy_gate.py

```
@@ -0,0 +1,547 @@
+                )
+            ),
+        },
+    ]
+
+
+def build_payload_without_mutations() -> dict[str, Any]:
+    policy_rows = build_policy_rows_without_final_accounting()
+    evaluated_rows = evaluate_rows(policy_rows)
+    selected = next(row for row in evaluated_rows if row["selected_without_final_accounting"])
+    if selected["variant_id"] != BEST_VARIANT_ID:
+        raise PredecommitOpeningPolicyGateError("selected variant drift")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)